tests/test_core.py

```diff
@@ -91,7 +92,7 @@ def test_stream_logs(self, openshift):
         (flexmock(openshift)
             .should_receive('_get')
              # First: timeout in response after 100s
-            .and_raise(httplib.IncompleteRead(''))
+            .and_raise(ChunkedEncodingError(''))
```

Comment:
I actually think we should have two test cases:
* some data is sent, then httplib.IncompleteRead is raised
* no data is sent, then ChunkedEncodingError is raised

I believe this is what's happening. When data has been sent but the remote endpoint closes the connection, that's when we get httplib.IncompleteRead. However if no data has been sent but the remote endpoint closes the connection, that's technically an error because the headers said to expect data -- so we get ChunkedEncodingError.